File: 69-sqrtx/69-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if(x==0)return 0;
        int low=1,right=x;
        int res;
        while(low<=right){
            int mid = low + (right-low)/2;
            if(mid==x/mid)return mid;
            else if(mid<x/mid){
                res= mid;
                low=mid+1;
            }
            else right=mid-1;
        }
        return res;
    }
};

File: 69-sqrtx/69-sqrtx.java
class Solution {
    public long ans(int x){
        if(x==1) return 1;
        long ans = -1;
        long start = 0 , end = x;
        while(start<= end){
            long mid = start + (end-start)/2;
            if(mid*mid<=x){
                ans = mid;
                start = mid+1;
            }
            else end = mid-1;
        }
        return ans;
        
    }
    public int mySqrt(int x) {
        long res = ans(x);
        return (int)res;
    }
}